Zend/zend_builtin_functions.c

```diff
@@ -491,12 +491,16 @@ ZEND_FUNCTION(func_num_args)
 {
 	zend_execute_data *ex = EX(prev_execute_data);
 
-	if (!(ZEND_CALL_INFO(ex) & ZEND_CALL_CODE)) {
-		RETURN_LONG(ZEND_CALL_NUM_ARGS(ex));
-	} else {
+	if (ZEND_CALL_INFO(ex) & ZEND_CALL_CODE) {
 		zend_error(E_WARNING, "func_num_args():  Called from the global scope - no function context");
 		RETURN_LONG(-1);
 	}
+
+	if (zend_forbid_dynamic_call("func_num_args()") == FAILURE) {
```

Comment:
hmm, I think maybe I didn't make myself clear,  I mean, you don't need
```
if (zend_forbid_dynamic_call("func_num_args()") == FAILURE) {
````

you could make it simpler by:
````
if (zend_forbid_dynamic_call() == FAILURE) {
````
the function name can be get via EG(current_execute_data).